skiko/src/jvmMain/kotlin/org/jetbrains/skiko/SkiaLayer.kt

```diff
@@ -135,11 +142,12 @@ open class SkiaLayer(
         )
     }
 
-    private fun fallbackToRaster() {
-        println("Falling back to software rendering...")
+    private fun fallbackToNextApi() {
+        val nextApi = fallbackRenderApiQueue.removeAt(0)
+        println("Falling back to $nextApi rendering...")
         redrawer?.dispose()
-        skijaState = SoftwareContextHandler(this)
-        redrawer = RasterRedrawer(this)
+        contextHandler = createContextHandler(this, nextApi)
```

Comment:
We should also call `contextHandler?.dispose()`

Reply:
Done.